lib/KTable/index.vue

```diff
@@ -0,0 +1,480 @@
+            ]"
+            role="gridcell"
+            :aria-colindex="colIndex + 1"
+            @keydown="handleKeydown($event, rowIndex, colIndex)"
+          >
+            <template #default="slotProps">
+
+              <!-- @slot Scoped slot for customizing the content of each data cell. Receives the following slot props - `content`: The content of the current cell,`rowIndex`: The index of the current row & `colIndex`: The index of the current column.-->
+              <slot name="cell" :content="slotProps.content" :rowIndex="rowIndex" :colIndex="colIndex" :row="row">
+                {{ slotProps.content }}
+              </slot>
+            </template>
+          </KTableGridItem>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+
+</template>
+
+
+<script>
+
+  import { ref, computed, watch } from '@vue/composition-api';
+  import useSorting, {
+    SORT_ORDER_ASC,
+    SORT_ORDER_DESC,
+    DATA_TYPE_OTHERS,
+  } from '../composables/useSorting';
+  import KTableGridItem from './KTableGridItem.vue';
+
+  export default {
+    name: 'KTable',
+    components: {
+      KTableGridItem,
+    },
+    setup(props, { emit }) {
+      const headers = ref(props.headers);
+      const rows = ref(props.rows);
+      const useLocalSorting = ref(props.useLocalSorting);
+      const {
+        sortKey,
+        sortOrder,
+        sortedRows,
+        handleSort: localHandleSort,
+        getAriaSort,
+      } = useSorting(headers, rows, useLocalSorting);
+
+      const finalRows = computed(() => {
+        if (props.sortable) {
+          return useLocalSorting.value ? sortedRows.value : rows.value;
+        } else {
+          return rows.value;
+        }
+      });
+
+      watch(
+        () => props.rows,
+        newRows => {
+          rows.value = newRows;
+        }
+      );
+
+      const handleSort = index => {
+        if (headers.value[index].dataType === DATA_TYPE_OTHERS) {
+          return;
+        }
+        if (useLocalSorting.value) {
+          localHandleSort(index);
+        } else {
+          emit(
+            'changeSort',
+            index,
+            sortOrder.value === SORT_ORDER_ASC ? SORT_ORDER_DESC : SORT_ORDER_ASC
+          );
+        }
+      };
+
+      const getHeaderStyle = header => {
+        const style = {};
+        if (header.minWidth) style.minWidth = header.minWidth;
+        if (header.width) style.width = header.width;
+        return style;
+      };
+
+      return {
+        sortKey,
+        sortOrder,
+        finalRows,
+        handleSort,
+        getAriaSort,
+        SORT_ORDER_ASC,
+        SORT_ORDER_DESC,
+        DATA_TYPE_OTHERS,
+        getHeaderStyle,
+      };
+    },
+    /* eslint-disable kolibri/vue-no-unused-properties */
+    props: {
+      /**
+       * An array of objects representing the headers of the table. Each header should have a label and a dataType.
```

Comment:
```suggestion
       * An array of objects `{ label, dataType, minWidth, width }`representing the headers of the table. The `dataType` can be one of `'string'`, `'numeric'`, `'date'`, or `'others'`. `label` and `dataType` are required. `minWidth` and `width` are optional.
```

Few tweaks for clarity and to make the generated documentation format nice.